clients/feeder/feeder.go

```diff
@@ -384,6 +400,26 @@ type ClassDefinition struct {
 	Program Program `json:"program"`
 }
 
+type ClassDefinition struct {
+	V0 *Cairo0Definition
+	V1 *SierraDefinition
+}
+
+func (c *ClassDefinition) UnmarshalJSON(data []byte) error {
+	jsonMap := make(map[string]any)
+	if err := json.Unmarshal(data, &jsonMap); err != nil {
+		return err
+	}
+
+	if _, found := jsonMap["sierra_program"]; found {
+		c.V1 = new(SierraDefinition)
+		return json.Unmarshal(data, c.V1)
+	} else {
+		c.V0 = new(Cairo0Definition)
+		return json.Unmarshal(data, c.V0)
+	}
```

Comment:
Pedantic [styling](https://github.com/uber-go/guide/blob/master/style.md#reduce-nesting) suggestion.
```suggestion
        }
        c.V0 = new(Cairo0Definition)
        return json.Unmarshal(data, c.V0)
```